### ISSUE TITLE
Saving issue with pdf backend

### ISSUE THREAD

**Issue by ericpre** [matplotlib/matplotlib#17599]
### Bug report

**Bug summary**

The animated artists are missing when saving a figure with the pdf backend (matplotlib 3.0.0 and more recent). The png backend is working fine.

**Code for reproduction**

```python
# Paste your code here
import matplotlib.pyplot as plt
import numpy as np

x = np.linspace(0, 2 * np.pi, 100)

fig, ax = plt.subplots()

(ln,) = ax.plot(x, np.sin(x), animated=True)
plt.show(block=False)
plt.pause(0.1)

bg = fig.canvas.copy_from_bbox(fig.bbox)
ax.draw_artist(ln)
fig.canvas.blit(fig.bbox)

plt.savefig('Trial.pdf')
```

**Actual outcome**

![image](https://user-images.githubusercontent.com/11851990/84190784-6208a380-aa8f-11ea-9cf2-3146a633e554.png)

**Expected outcome**
![image](https://user-images.githubusercontent.com/11851990/84190809-6a60de80-aa8f-11ea-85ea-cac043157128.png)

Saving using the following works fine:

```python
# This is working fine
from matplotlib.backends.backend_pdf import PdfPages
with PdfPages('Trial.pdf') as pdf:
     pdf.savefig()
```

**Matplotlib version**

  * Operating system: linux
  * Matplotlib version: 3.0.0 and above
  * Matplotlib backend (`print(matplotlib.get_backend())`): 'Qt5Agg'
  * Python version: 3.7.7
  * Jupyter version (if applicable):
  * Other libraries:

**Comment by tacaswell** [comment]
Thanks for this report!   It ended up being a simple fix, can you test the branch in #17602?

**Comment by ericpre** [comment]
Thanks for the quick fix, the other example in https://github.com/hyperspy/hyperspy/issues/2374 is working as expected using matplotlib master branch.
